syncer/syncer.go

```diff
@@ -395,25 +395,21 @@ func (s *Syncer) Init() (err error) {
 	}
 	rollbackHolder.Add(fr.FuncRollback{Name: "remove-active-realylog", Fn: s.removeActiveRelayLog})
 
-	// init successfully, close done chan to make Syncer can be closed
-	// when Process started, we will re-create done chan again
-	// NOTE: we should refactor the Concurrency Model some day
-	s.done = make(chan struct{})
-	close(s.done)
+	s.reset()
```

Comment:
seems the purpose of `reset` is not very clear, such as can we put these two lines into `s.reset`? btw is `s.reset()` necessary here?
```
s.execErrorDetected.Set(false)
s.resetExecErrors()
```

Reply:
yes, it's lost two lines. I fixed it.
if we don't put `s.reset()` here,  where is better? if we put it in `Process`, one question is that we start `Process` in another goroutine, but the some part of `s.reset()` like `	s.ddlExecInfo.Renew()` is better done in main goroutine, otherwise we need to sleep some seconds to ensure `s.ddlExecInfo.Renew()` finished.

Reply:
yep, you’re right